### DISCUSSION TITLE
why can't Github allow for access restriction on artifacts (GITHUB ACTIONS)

### DISCUSSION THREAD

**Discussion by brokedba** [community/community#54216] [Actions]
### Select Topic Area

Product Feedback

### Select Feature Area

- [X] Issues
- [ ] Projects

### Body

#foodforThought
From the API documentation I could read the below   : 
[Download an artifact #](https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#download-an-artifact)
Works with [GitHub Apps](https://docs.github.com/en/developers/apps)
Gets a redirect URL to download an archive for a repository. This URL expires after 1 minute. Look for Location: in the response header to find the URL for the download. The :archive_format must be zip. **Anyone with read access to the repository can use this endpoint**. If the repository is private you must use an access token with the repo scope. GitHub Apps must have the actions:read permission to use this endpoint.

- Why can't Github allow to have a default policy that allows repository owners to decide who has read access on the artifacts be it in public or private repository?
- There are so many cases where users would love to use artifact I Github actions workflow but can't because the artifact contains sensitive content .  
- Having the possibility to restrict external access to these artifact will make it safer to use it in our workflows. because for now the only alternative is using caches.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @brokedba ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
